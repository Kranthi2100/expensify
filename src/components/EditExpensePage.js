import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, startRemoveExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component{
  onClick(){
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/dashboard');
  }
  onSubmit(expense){
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/dashboard');
  }
  render(){
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={ this.onSubmit.bind(this) } 
        />
        <button onClick={ this.onClick.bind(this) }>Remove</button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
};

const mapDispatchToProps = (dispatch) =>{
  return {
    removeExpense : (data) =>{
      return dispatch(startRemoveExpense(data));
    },
    editExpense : (id, expense)  => {
      return dispatch(startEditExpense(id, expense));
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
