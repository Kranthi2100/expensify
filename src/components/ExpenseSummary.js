import React from 'react';
import getExpensesTotal from '../selectors/getExpensesTotal';
import numeral from 'numeral';
import { connect } from 'react-redux';
import getVisibleExpenses from '../selectors/expenses';
export const ExpenseSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>viewing {expensesCount} {expenseWord} totalling {formatedExpensesTotal}</h1>
        </div>
    );
}

const mapStateToProps = (state)=>{
    const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expensesTotal : getExpensesTotal(visibleExpenses),
        expensesCount : visibleExpenses.length
    }
}
export default connect(mapStateToProps)(ExpenseSummary);