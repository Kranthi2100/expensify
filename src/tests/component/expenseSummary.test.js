import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpenseSummary} from '../../components/ExpenseSummary';

test('should render expense summary', ()=>{
    const wrapper = shallow(<ExpenseSummary
            expensesCount={1} expensesTotal={353}
        />);
    expect(wrapper).toMatchSnapshot();
})

test('should render expense summary with multiple expenses', ()=>{
    const wrapper = shallow(<ExpenseSummary
            expensesCount={23} expensesTotal={354353453}
        />);
    expect(wrapper).toMatchSnapshot();
})