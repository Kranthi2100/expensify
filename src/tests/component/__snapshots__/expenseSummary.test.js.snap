// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense summary 1`] = `
<div>
  <h1>
    viewing 
    1
     
    expense
     totalling 
    $3.53
  </h1>
</div>
`;

exports[`should render expense summary with multiple expenses 1`] = `
<div>
  <h1>
    viewing 
    23
     
    expenses
     totalling 
    $3,543,534.53
  </h1>
</div>
`;
